infra/feast-operator/api/v1alpha1/featurestore_types.go

```diff
@@ -71,16 +73,65 @@ type FeatureStoreServices struct {
 // OfflineStore configures the deployed offline store service
 type OfflineStore struct {
 	ServiceConfigs `json:",inline"`
+	Persistence    *OfflineStorePersistence `json:"persistence,omitempty"`
+}
+
+// OfflineStorePersistence configures the persistence settings for the offline store service
+type OfflineStorePersistence struct {
+	FilePersistence *OfflineStoreFilePersistence `json:"file,omitempty"`
+}
+
+// OfflineStorePersistence configures the file-based persistence for the offline store service
+type OfflineStoreFilePersistence struct {
+	// +kubebuilder:validation:Enum=dask;duckdb
+	Type string `json:"type,omitempty"`
+}
+
+var validOfflineStoreFilePersistenceType = []string{
+	"dask",
+	"duckdb",
+}
+
+// A function to validate the file persistence types for offline stores
+func IsValidOfflineStoreFilePersistenceType(value string) (bool, error) {
```

Comment:
let's move this function and its corresponding var to the `services` section of code instead, and run it early on during the `feast.Deploy()`

Reply:
its also worth noting that this should never be an issue as long as the CRD schema validation remains intact. this new `IsValidOfflineStoreFilePersistenceType` function would be backup validation for an extreme edge case.